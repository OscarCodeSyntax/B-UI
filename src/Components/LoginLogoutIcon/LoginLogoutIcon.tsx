import { Login, Logout } from "@mui/icons-material";
import React from "react";
import isLoggedInConverter from "../../Resources/Helpers/IsLoggedInConverter";
import HandleLogout from "../../Resources/Helpers/HandleLogout";

function LoginLogoutIcon(isLoggedIn: string) {
  const LoginLogoutIcon = isLoggedInConverter(isLoggedIn) ? (
    <div className="split padding-20">
      <a
        className="flex color-black"
        href="/"
        onClick={(e: any) => HandleLogout(e)}
      >
        <div>
          <div className="margin-5"> Logout</div>
        </div>
        <Logout fontSize="large" className="margin-login" />
      </a>
    </div>
  ) : (
    <div className="split padding-20">
      <a
        className="flex color-black"
        href="/"
        onClick={(e: any) => HandleLogout(e)}
      >
        <div>
          <div className="margin-5"> Login</div>
        </div>
        <Login fontSize="large" className="margin-login " />
      </a>
    </div>
  );

  return LoginLogoutIcon;
}

export default LoginLogoutIcon;
