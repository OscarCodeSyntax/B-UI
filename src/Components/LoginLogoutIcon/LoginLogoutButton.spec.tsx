import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import LoginLogoutIcon from "./LoginLogoutIcon";

describe("LoginLogoutIcon ", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("renders correctly with true prop", () => {
    const LoginLogoutIconWrapper = renderer
      .create(LoginLogoutIcon("true"))
      .toJSON();
    expect(LoginLogoutIconWrapper).toMatchSnapshot();
  });

  it("renders correctly with false prop", () => {
    const LoginLogoutIconWrapper = renderer
      .create(LoginLogoutIcon("false"))
      .toJSON();
    expect(LoginLogoutIconWrapper).toMatchSnapshot();
  });

  it("renders correct link, icon and text when true", () => {
    const testRenderer = renderer.create(LoginLogoutIcon("true"));
    const testInstance = testRenderer.root;

    expect(testInstance.findAllByType("a").length).toEqual(1);
    expect(testInstance.findAllByType("svg").length).toEqual(1);
    expect(testInstance.findAllByType("div").length).toEqual(3);
  });

  it("renders correct link, icon and text when false", () => {
    const testRenderer = renderer.create(LoginLogoutIcon("false"));
    const testInstance = testRenderer.root;

    expect(testInstance.findAllByType("a").length).toEqual(1);
    expect(testInstance.findAllByType("svg").length).toEqual(1);
    expect(testInstance.findAllByType("div").length).toEqual(3);
  });

  it("has all the correct text when true", () => {
    render(LoginLogoutIcon("true"));
    expect(screen.getByText("Logout")).toBeTruthy();
    expect(screen.getByTestId("LogoutIcon")).toBeTruthy();
  });

  it("has all the correct text when true", () => {
    render(LoginLogoutIcon("false"));
    expect(screen.getByText("Login")).toBeTruthy();
    expect(screen.getByTestId("LoginIcon")).toBeTruthy();
  });
});
