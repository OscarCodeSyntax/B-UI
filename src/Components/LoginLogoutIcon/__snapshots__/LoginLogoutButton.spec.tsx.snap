// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginLogoutIcon  renders correctly with false prop 1`] = `
<div
  className="split padding-20"
>
  <a
    className="flex color-black"
    href="/"
    onClick={[Function]}
  >
    <div>
      <div
        className="margin-5"
      >
         Login
      </div>
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge margin-login  css-tzssek-MuiSvgIcon-root"
      data-testid="LoginIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8z"
      />
    </svg>
  </a>
</div>
`;

exports[`LoginLogoutIcon  renders correctly with true prop 1`] = `
<div
  className="split padding-20"
>
  <a
    className="flex color-black"
    href="/"
    onClick={[Function]}
  >
    <div>
      <div
        className="margin-5"
      >
         Logout
      </div>
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge margin-login css-tzssek-MuiSvgIcon-root"
      data-testid="LogoutIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4z"
      />
    </svg>
  </a>
</div>
`;
